iotdb-client/session/src/main/java/org/apache/iotdb/session/Session.java

```diff
@@ -2737,6 +2774,78 @@ private void updateTSInsertTabletsReq(
     request.addToSizeList(tablet.rowSize);
   }
 
+  public void invertToTabletsAndInsert(
+      List<String> deviceIds,
+      List<Long> times,
+      List<List<String>> measurementsList,
+      List<List<TSDataType>> typesList,
+      List<List<Object>> valuesList,
+      boolean isAligned)
+      throws IoTDBConnectionException, StatementExecutionException {
+    while (!deviceIds.isEmpty()) {
```

Comment:
Even if all the records are about the same measurement under the same device, if the order of measurements in each record is different, it could lead to one set of records being split into multiple tablets for transmission. In conditions of average network quality, the multiple RPC delays could result in high latency and potentially poor performance. I lean towards placing all records of a device into one tablet to complete the transmission in a single step.